modules/datepicker/basepicker.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react'
+import {Button} from 'react-native'
+
+import {useTheme} from '@frogpond/app-theme'
+import type {AppTheme} from '@frogpond/app-theme'
+
+import moment from 'moment-timezone'
+import type {Moment} from 'moment-timezone'
+
+import DateTimePicker from '@react-native-community/datetimepicker'
+import type {Event} from '@react-native-community/datetimepicker'
+
+import {BaseDatetimePickerProps} from './types'
+
+const FORMATS = {
+	date: 'YYYY-MM-DD',
+	datetime: 'YYYY-MM-DD HH:mm',
+	time: 'HH:mm',
+}
+
+export const BaseDateTimePicker = (
+	props: BaseDatetimePickerProps,
+): JSX.Element => {
+	let [date, setDate] = React.useState(props.initialDate)
+	let [timezone] = React.useState(props.initialDate.tz() || '')
+
+	const onChange = (_event: Event, selectedDate?: Date | undefined) => {
+		// Reset Android's modal's presenting state
+		props.onChange?.()
+
+		if (!selectedDate) {
+			return
+		}
+
+		setDate(moment(selectedDate))
+		props.onDateChange(moment.tz(selectedDate, timezone))
+	}
+
+	const formatDate = (date: Moment): string => {
+		const {mode, format = FORMATS[mode]} = props
```

Comment:
This type error is new, I think.

Here I believe `mode` is getting inferred to be `undefined`.

Reply:
Inside `export type IOSNativeProps` (inside of the dependency), we have

```typescript
    /**
     * The date picker mode.
     */
    mode?: IOSMode;
```

and the same in `AndroidNativeProps`.

Reply:
If I try specifying a default,

```diff
diff --git a/modules/datepicker/basepicker.tsx b/modules/datepicker/basepicker.tsx
index b8bfc1eea..f6949bce6 100644
--- a/modules/datepicker/basepicker.tsx
+++ b/modules/datepicker/basepicker.tsx
@@ -37,7 +37,7 @@ export const BaseDateTimePicker = (
        }

        const formatDate = (date: Moment): string => {
-               const {mode, format = FORMATS[mode]} = props
+               const {mode, format = FORMATS[mode || 'date']} = props
                return moment.tz(date, timezone).format(format)
        }
```

then I get a new type error, this one a bit more serious:

```diff
+modules/datepicker/basepicker.tsx(40,25): error TS7053: Element implicitly has an 'any' type because expression of type '"time" | "date" | "datetime" | "countdown"' can't be used to index type '{ date: string; datetime: string; time: string; }'.
+  Property 'countdown' does not exist on type '{ date: string; datetime: string; time: string; }'.
```

So there are probably a couple of routes we could go.

- If we only want to allow `date`, `datetime`, and `time` types, we should encode that into the type definition:
  
  ```diff
  diff --git a/modules/datepicker/types.ts b/modules/datepicker/types.ts
  index 329e90705..796ee2bbf 100644
  --- a/modules/datepicker/types.ts
  +++ b/modules/datepicker/types.ts
  @@ -8,7 +8,8 @@ export interface BaseDatetimePickerProps {
          // dep-specific
          initialDate: Moment
          minuteInterval?: 1 | 2 | 3 | 4 | 5 | 6 | 10 | 12 | 15 | 20 | 30
  -       mode: IOSNativeProps['mode'] | AndroidNativeProps['mode']
  +       mode: (IOSNativeProps['mode'] | AndroidNativeProps['mode']) &
  +               ('date' | 'datetime' | 'time')
          format?: string
          onDateChange: (moment: Moment) => void
  ```
  
  Since doing that also removes the possibility for the value to be `undefined`, you wouldn't need to specify a variant.
  
  We could probably do something with enums or just encode the list of supported format keys as an array somewhere so we don't have to specify this inline.
  
- Otherwise, we should figure out how to make `FORMATS` keys match up one-to-one with the IOSNativeProps and AndroidNativeProps things. This would mean we have support for the other types, but would require a bit more effort, I think.

Reply:
I’m super thrilled you picked up on this and appreciate the multiple avenues for resolving it.

Also I think at one point I thought about using `Pick` to choose the modes explicitly (since `Exclude` could lead to ones being added that we didn’t approve of before).